src/main/java/io/supertokens/inmemorydb/queries/ThirdPartyQueries.java

```diff
@@ -172,19 +162,40 @@ public static UserInfo getUserInfoUsingId_Transaction(Start start, Connection co
             pst.setLong(2, timeJoined);
             pst.setString(3, userId);
             pst.setInt(4, limit);
-            ResultSet result = pst.executeQuery();
-            List<UserInfo> temp = new ArrayList<>();
-            while (result.next()) {
-                temp.add(UserInfoRowMapper.getInstance().mapOrThrow(result));
-            }
-            UserInfo[] finalResult = new UserInfo[temp.size()];
-            for (int i = 0; i < temp.size(); i++) {
-                finalResult[i] = temp.get(i);
-            }
-            return finalResult;
+
+            return resultToArray(pst.executeQuery());
         }
     }
 
+    public static UserInfo[] getThirdPartyUsersByEmail(Start start, @Nonnull String email)
+        throws SQLException, StorageQueryException {
+
+        String sqlQuery = "SELECT user_id, third_party_id, third_party_user_id, email, time_joined FROM " +
+                Config.getConfig(start).getThirdPartyUsersTable() +
+                " WHERE email = ?" +
+                " ORDER BY user_id DESC";
+
+        if (email.equals("*")) {
```

Comment:
Input sanitization should happen in the API layer, not here. There, you should check that the input email correctly matches the format of an email. Anyway, with prepared statements using `*` as email will just yield an empty result.